Sources/ApodiniNetworking/URLQueryParamDecoder.swift

```diff
@@ -10,69 +10,78 @@ import Foundation
-                self.decodingStrategyDesc = {
-                    switch decodingStrategy {
-                    case .iso8601: return "iso8601"
-                    case .secondsSince1970: return "secondsSince1970"
-                    case .secondsSinceReferenceDate: return "secondsSinceReferenceDate"
-                    case .custom: return "custom"
-                    }
-                }()
-            }
-            
-            var errorDescription: String? {
-                "Unable to decode date from raw input '\(rawInput)'. Used strategy: \(decodingStrategyDesc)"
-            }
-        }
-        
-        case iso8601
-        case secondsSince1970
-        case secondsSinceReferenceDate
-        case custom((String) throws -> Date)
+/// How a decoder should decode dates.
+public enum DateDecodingStrategy {
+    struct DateDecodingError: Error, LocalizedError {
+        let rawInput: String
+        let decodingStrategyDesc: String
         
-        func decodeDate(from rawValue: String) throws -> Date {
-            switch self {
-            case .iso8601:
-                if let date = ISO8601DateFormatter().date(from: rawValue) {
-                    return date
-                } else {
-                    throw DateDecodingError(rawInput: rawValue, decodingStrategy: self)
+        init(rawInput: String, decodingStrategy: DateDecodingStrategy) {
+            self.rawInput = rawInput
+            self.decodingStrategyDesc = {
+                switch decodingStrategy {
+                case .iso8601: return "iso8601"
+                case .secondsSince1970: return "secondsSince1970"
+                case .secondsSinceReferenceDate: return "secondsSinceReferenceDate"
+                case .custom: return "custom"
                 }
-            case .secondsSince1970:
-                guard let numericValue = TimeInterval(rawValue) else {
-                    throw DateDecodingError(rawInput: rawValue, decodingStrategy: self)
-                }
-                return Date(timeIntervalSince1970: numericValue)
-            case .secondsSinceReferenceDate:
-                guard let numericValue = TimeInterval(rawValue) else {
-                    throw DateDecodingError(rawInput: rawValue, decodingStrategy: self)
-                }
-                return Date(timeIntervalSinceReferenceDate: numericValue)
-            case .custom(let decodingFn):
-                return try decodingFn(rawValue)
-            }
+            }()
+        }
+        
+        var errorDescription: String? {
+            "Unable to decode date from raw input '\(rawInput)'. Used strategy: \(decodingStrategyDesc)"
         }
     }
     
+    /// Decodes `Date` objects from an ISO8601 string
+    case iso8601
+    /// Decodes `Date` objects from a UNIX timestamp
+    case secondsSince1970
+    /// Decodes `Date` objects from a timestamp relative to Foundation's reference date of 00:00:00 UTC on 1 January 2001.
+    case secondsSinceReferenceDate
```

Comment:
Maybe a more meaningful name? Suggestion: `secondsSinceFoundationReferenceDate`

Reply:
reference date is fine, since that's a well-established term familiar to anyone who ever worked with Apple's date APIs. Also, it's explained in the documentation.